politeiawww/api/cms/v1/v1.go

```diff
@@ -513,3 +564,68 @@ type EditUser struct {
 
 // EditUserReply is the reply for the EditUser command.
 type EditUserReply struct{}
+
+// DCCInput contains all of the information concerning a DCC object that
+// will be submitted as a Record to the politeiad backend.
+type DCCInput struct {
+	Type             DCCTypeT    `json:"type"`          // Type of DCC object
+	NomineeUserID    string      `json:"nomineeuserid"` // UserID of the DCC nominee (issuance or revocation)
+	SponsorStatement string      `json:"statement"`     // Statement from sponsoring user about why DCC should be approved
+	Domain           DomainTypeT `json:"domain"`        // Domain of proposed contractor issuance
+}
+
+// DCCRecord is what will be decoded from a Record for a DCC object to the politeiad backend.
+type DCCRecord struct {
+	Type               DCCTypeT   `json:"type"`               // Type of DCC
+	Status             DCCStatusT `json:"status"`             // Current status of the DCC
+	StatusChangeReason string     `json:"statuschangereason"` // The reason for changing the DCC status.
+	Timestamp          int64      `json:"timestamp"`          // Last update of dcc
+	DCC                DCCInput   `json:"dccpayload"`         // DCC payload for the given object
+	Files              []www.File `json:"file"`               // Actual DCC files (dcc.json, etc)
+	PublicKey          string     `json:"publickey"`          // Sponsoring user's public key, used to verify signature.
+	Signature          string     `json:"signature"`          // Signature of file digest
+	Version            string     `json:"version"`            // Record version
+
+	SponsorStatement string      `json:"statement"` // Sponsor statement in support of the DCC
+	Domain           DomainTypeT `json:"domain"`    // Domain of DCC
+
+	SponsorUserID   string `json:"sponsoruserid"`   // The userid of the sponsoring user.
+	SponsorUsername string `json:"sponsorusername"` // The username of the sponsoring user.
+
+	SupportUserIDs    []string `json:"supportuserids"` // List of UserIDs for those that have shown support of the DCC.
+	OppositionUserIDs []string `json:"againstuserids"` // List of UserIDs for those that have shown opposition of the DCC.
+
+	CensorshipRecord www.CensorshipRecord `json:"censorshiprecord"`
+}
+
+// NewDCC is a request for submitting a new DCC proposal.
+type NewDCC struct {
+	Files     []www.File `json:"files"`     // Issuance/Revocation file and any attachments along with it
```

Comment:
I think we should only allow for a single file, the dcc.json file, to be submitted.  If there comes a point in time where there is a clear use case for allowing attachments then it can be added.